src/iamserver/nodecontroller.cpp

```diff
@@ -355,10 +323,33 @@ void NodeController::Close()
 grpc::Status NodeController::HandleRegisterNodeStream(const std::vector<aos::NodeStatus>& allowedStatuses,
     NodeServerReaderWriter* stream, grpc::ServerContext* context, aos::iam::nodemanager::NodeManagerItf* nodeManager)
 {
-    auto handler = std::make_shared<NodeStreamHandler>(allowedStatuses, stream, context, nodeManager);
-    StoreHandler(handler);
+    auto handler = std::make_shared<NodeStreamHandler>(stream, context);
+
+    auto onNodeInfoReceivedCallback
+        = [this, nodeManager, handler, &allowedStatuses](const iamproto::NodeInfo& info) -> aos::Error {
```

Comment:
is it guaranteed that for example `allowedStatuses` will be exist? maybe use it as private method or another case ? or this callback sync and it will not left this method ?